zipline/transforms/batch_transform.py

```diff
@@ -486,11 +487,13 @@ def _get_field_names(self, event):
             return set.union(self.field_names, self.latest_names)
 
 
+@deprecated(msg="batch_transform is deprecated and will be removed"
```

Comment:
Should this live on `BatchTransform` instead?  Not sure if there are other pathways to it.